akka-actor-tests/src/test/scala/akka/pattern/CircuitBreakerMTSpec.scala

```diff
@@ -36,52 +36,69 @@ class CircuitBreakerMTSpec extends AkkaSpec {
       awaitCond(failingCall())
     }
 
-    def testCallsWithBreaker(breaker: CircuitBreaker): immutable.IndexedSeq[Future[String]] = {
-      val aFewActive = new TestLatch(5)
-      for (_ <- 1 to numberOfTestCalls)
-        yield breaker
-          .withCircuitBreaker(Future {
-            aFewActive.countDown()
-            Await.ready(aFewActive, 5.seconds.dilated)
-            "succeed"
-          })
-          .recoverWith {
-            case _: CircuitBreakerOpenException =>
-              aFewActive.countDown()
-              Future.successful("CBO")
-          }
+    def testCallsWithBreaker(breaker: CircuitBreaker, numberOfCalls: Int): immutable.IndexedSeq[Future[String]] = {
+      for (_ <- 1 to numberOfCalls)
+        yield makeCallWithBreaker(breaker)
     }
 
+    def makeCallWithBreaker(breaker: CircuitBreaker): Future[String] =
+      breaker.withCircuitBreaker(Future.successful("succeed")).recoverWith {
+        case _: CircuitBreakerOpenException =>
+          Future.successful("CBO")
+      }
```

Comment:
in most case, that's all that we need. Either it `succeed` or it recovered with `CBO`